tmpl/wiki/hello-world.md

```diff
@@ -1,232 +1,486 @@
 First make sure you have followed the [installation instructions](/wiki/install)
-to get a working MirageOS installation. The examples below are in the
-[mirage-skeleton](http://github.com/mirage/mirage-skeleton) repository. Begin by
-cloning and changing directory to it:
+to get a working MirageOS installation. The examples below are in
+the [mirage-skeleton](http://github.com/mirage/mirage-skeleton) repository.
+Begin by cloning and changing directory to it:
 
 ```
 $ git clone git://github.com/mirage/mirage-skeleton.git
 $ cd mirage-skeleton
 ```
 
+The `mirage-skeleton` repository classifies its examples into three groups.
+This document refers to unikernels in the `tutorial` directory.
+
+**Note**: Before we begin, if you aren't familiar with the Lwt library (and the
+`>>=` operator it provides), you may want to read at least the start of
+the [Lwt tutorial](tutorial-lwt) first.
+
+**Additional note**: Throughout the tutorial, we'll use `mirage configure -t unix`
+to demonstrate building MirageOS applications.  If you're using a Mac OS X
+machine, you should use `mirage configure -t macosx` instead.
+
+### Step 0: Doing Nothing!
+
+Before we try and do anything complicated, let's do nothing briefly. That is,
+let's build a unikernel that simply starts and then exits -- nothing else. The
+code for this is, as you might hope, fairly short. First the unikernel itself:
+
+```
+$ cat tutorial/noop/unikernel.ml
+let start =
+  Lwt.return_unit
+```
+
+So this is a unikernel whose entry point (`start`) does nothing other than
+return an `Lwt` thread that will evaluate to `unit`. Easy.
+
+Before we can build even our `noop` unikernel, we must define its configuration.
+That is, we need to tell Mirage what OCaml module contains the `start` entry
+point. We do this by writing a `config.ml` file that sits next to our
+`unikernel.ml` file (although you can name the file containing the configuration
+something else, the `mirage` tool defaults to `config.ml`):
```

Comment:
"(although you can name the file containing the configuration something else" <- not true anymore!?